python/triton/compiler/compiler.py

```diff
@@ -345,6 +345,36 @@ def get_architecture_descriptor(capability):
     return capability
 
 
+def get_architecture_num_warps(device_type, capability=None):
+    if device_type in ["cuda", "hip"]:
+        arch = get_architecture_descriptor(capability)
+    else:
+        _device_backend = get_backend(device_type)
+        assert _device_backend
+        arch = _device_backend.get_architecture_descriptor(device_type=device_type, capability=capability)
+
+    is_cuda = device_type == "cuda" and _is_cuda(arch)
```

Comment:
Why cannot we just use `device_type == "cuda"`? I think I'm missing something. Please remind me if any.

Reply:
Maybe in some cases the `cuda` here also means the AMD GPU. In the torch the AMD GPU uses the the `cuda` as well.

I am not 100% sure about this. So I kept this as same as the line here:
https://github.com/openai/triton/blob/812bc5e05381a84f46157cd5fa211238c6b42f20/python/triton/compiler/compiler.py#L416